bokeh/sphinxext/bokeh_plot.py

```diff
@@ -241,7 +243,15 @@ def _process_script(source, filename, env, js_name, use_relative_paths=False):
 
     c = ExampleHandler(source=run_source, filename=filename)
     d = Document()
-    c.modify_document(d)
+
+    # We may need to instantiate deprecated objects as part of documenting
+    # them in the reference guide. Suppress any warnings here to keep the
+    # docs build clean just for this case
+    with warnings.catch_warnings():
+        if "reference" in env.docname:
```

Comment:
This definitely feels sketchy. Where does `"reference"` come from?

Reply:
This comes from the path to the .rst file:
```
(Pdb) env.docname
'docs/reference/plotting'
```
I don't think it's perfect, but I do think it is good enough.

Reply:
I see. I guess it should be obvious to someone that knows what `env.docname` contains. :) I couldn't find anything quickly.

Reply:
Right :) The small risk is that some non-reference-guide file has "reference" in the path, and also uses a deprecated thing. In the worst case the deprecated thing is in the docs until it is is removed (at which point it will have to be since the docs would break at that point otherwise) and I can live with that. My *preference* is that deprecated things be confined to the ref guide, but I'm not going to sweat this small-risk possibility that is also ultimately self-resolving.